src/cublaslt/safe.rs

```diff
@@ -85,6 +85,194 @@ pub enum Activation {
     Gelu,
 }
 
+/// MatrixLayout helper type
+struct MatrixLayout {
+    handle: sys::cublasLtMatrixLayout_t,
+}
+
+impl MatrixLayout {
+    fn new(
+        matrix_type: sys::cudaDataType,
+        rows: u64,
+        cols: u64,
+        ld: i64,
+    ) -> Result<Self, CublasError> {
+        let handle = result::create_matrix_layout(matrix_type, rows, cols, ld)?;
+        Ok(Self { handle })
+    }
+
+    fn set_batch(&self, size: c_int, stride: i64) -> Result<(), CublasError> {
+        unsafe {
+            // Set batch size
+            set_matrix_layout_attribute(
+                self.handle,
+                sys::cublasLtMatrixLayoutAttribute_t::CUBLASLT_MATRIX_LAYOUT_BATCH_COUNT,
+                (&size) as *const _ as *const _,
+                mem::size_of::<c_int>(),
+            )?;
+            // Set batch stride
+            set_matrix_layout_attribute(
+                self.handle,
+                sys::cublasLtMatrixLayoutAttribute_t::CUBLASLT_MATRIX_LAYOUT_STRIDED_BATCH_OFFSET,
+                (&stride) as *const _ as *const _,
+                mem::size_of::<i64>(),
+            )?;
+        }
+        Ok(())
+    }
+}
+
+impl Drop for MatrixLayout {
+    fn drop(&mut self) {
+        // panic on failure
+        unsafe {
+            result::destroy_matrix_layout(self.handle).expect("Unable to destroy matrix layout")
```

Comment:
I'm potentially panicking on drop. Is that the correct way of dealing with failable functions in drop?

Reply:
This is how it's done elsewhere indeed.